packages/sanity/src/core/preview/useLiveDocumentIdSet.ts

```diff
@@ -0,0 +1,45 @@
+import {type QueryParams} from '@sanity/client'
+import {useMemo} from 'react'
+import {useObservable} from 'react-rx'
+import {scan} from 'rxjs/operators'
+
+import {useDocumentPreviewStore} from '../store/_legacy/datastores'
+import {type DocumentIdSetObserverState} from './liveDocumentIdSet'
+
+const INITIAL_STATE = {status: 'loading' as const, documentIds: []}
+
+export type LiveDocumentSetState =
+  | {status: 'loading'; documentIds: string[]}
+  | DocumentIdSetObserverState
+
+/**
+ * @internal
+ * @beta
+ *
+ * Observes a document by its ID and returns the document and loading state
```

Comment:
Q: I'm struggling to follow how this returns the document, and not just the ID of the doc? I can see `unstable_observeDocument` [used here](https://github.com/sanity-io/sanity/pull/7253/files#diff-dbf3eecd45532983c3957c1581980b927a5fb82d9597b3e366fd4d9223082ff9R27) and was expecting something similar in this hook to return the full document

Reply:
Gah, copy paste error, thanks for catching!

Reply:
Fixed by 17ab291a61dc8c79e253ccf562d30b6f80be2a4e